hbogolib/handlersp.py

```diff
@@ -474,7 +504,11 @@ def addDir(self, item, mode=1, media_type=None):
     def addCat(self, name, url, icon, mode):
         if self.lograwdata:
             self.log("Adding Cat: " + str(name) + "," + str(url) + "," + str(icon) + " MODE: " + str(mode))
-        u = self.base_url + "?url=" + quote(url) + "&mode=" + str(mode) + "&name=" + quote(name)
+        u = '%s?%s' % (self.base_url, {
```

Comment:
There is an error here needs to be substituted with:

```
u = '%s?%s' % (self.base_url, urlencode({
            'url': url,
            'mode': mode,
            'name': name}))
```

Reply:
Fixed